src/controllers/deliveryLog.js

```diff
@@ -1,24 +1,59 @@
+import { getCohortById } from '../domain/cohort.js'
+import { createDeliveryLogDb } from '../domain/deliveryLog.js'
 import { sendDataResponse } from '../utils/responses.js'
+import ERR from '../utils/errors.js'
 
-export const create = async (req, res) => {
-  const { date, cohort_id: cohortId, lines } = req.body
+export const createDeliveryLog = async (req, res) => {
+  const { cohort_id: cohortId, lines } = req.body
+  const { firstName, lastName } = req.user
 
-  return sendDataResponse(res, 201, {
-    log: {
-      id: 1,
-      cohort_id: cohortId,
-      date,
-      author: {
-        id: req.user.id,
-        first_name: req.user.firstName,
-        last_name: req.user.lastName
-      },
-      lines: lines.map((line, index) => {
-        return {
-          id: index + 1,
-          content: line.content
+  try {
+    if (!cohortId || !lines) {
+      throw Error(ERR.INCOMPLETE_REQUEST)
+    }
+
+    const cohortExists = await getCohortById(cohortId)
+    if (!cohortExists) {
+      throw Error(ERR.COHORT_NOT_FOUND)
+    }
+
+    const log = await createDeliveryLogDb(cohortId, req.user.id, lines)
+
+    return sendDataResponse(res, 201, {
+      data: {
+        log: {
+          id: log.id,
+          cohortId: cohortId,
+          date: log.date,
+          author: {
+            id: log.user.id,
+            firstName,
+            lastName
+          },
+          lines: log.lines.map((line) => ({
+            id: line.id,
+            content: line.content
+          }))
         }
+      }
+    })
+  } catch (error) {
+    console.error(error)
```

Comment:
Add some context to this error message

Reply:
What you mean by this? It's a geric error. Server died or something. :D

Reply:
Something like `console.error('error creating delivery log', error)`